File: golang/go-demo/array/array.go
package main

import "fmt"

func main() {
	// declaring arrays and slices

	// array elements is initialized as default values
	arr1 := [5]int{}
	// specifying elements values
	// declaring slices without specifying arrays' length
	arr2 := []int{1, 2, 3, 4, 5}
	arr3 := [...]int{1, 2, 3, 4, 5}
	// initializing some elements, others are default values
	arr4 := [10]int{1, 2, 3, 4, 5}

	// using slices as parameters
	printArray := func(arr []int) {
		for i := 0; i < len(arr); i++ {
			fmt.Printf("%d ", arr[i])
		}
		fmt.Println()
	}
	printArray(arr1[:])
	printArray(arr2)
	printArray(arr3[:])
	printArray(arr4[:])
}

File: php/php-sample/types/array.php
<?php

# in php arrays are actually hashmaps
$arr1 = [];
$arr2 = array();
$arr3 = [1, 2, 3,];
$arr4 = [1 => 'a', "2" => 3, 'abc' => 4, 5];

var_dump($arr4);

# accessing array elements
assert(1 == $arr3[1]);

# deleting elements
unset($arr4['abc']);



